src/client.ts

```diff
@@ -1,499 +1,424 @@
+      this.queryClient.oracleScript(
+        request,
+        {} as grpc.Metadata,
+        (err, response) => {
+          if (err !== null) {
+            reject(err)
+            return
+          }
+
+          if (response !== null && response.hasOracleScript()) {
+            resolve(response.getOracleScript().toObject())
+          }
+        },
+      )
+    })
+  }
 
-  async getOracleScript(id: number): Promise<OracleScript> {
+  async getRequestById(id: number): Promise<QueryRequestResponse.AsObject> {
     if (!Number.isInteger(id)) throw new NotIntegerError('id is not an integer')
 
-    const response = await this.getResult(`/oracle/oracle_scripts/${id}`)
-    return {
-      owner: Address.fromAccBech32(response.owner),
-      name: response.name,
-      description: response.description,
-      fileName: response.filename,
-      schema: response.schema,
-      sourceCodeUrl: response.source_code_url,
-    }
+    const request = new QueryRequestRequest()
+    request.setRequestId(id)
+    return new Promise((resolve, reject) => {
+      this.queryClient.request(
+        request,
+        {} as grpc.Metadata,
+        (err, response) => {
+          if (err !== null) {
+            reject(err)
+            return
+          }
+
+          if (response !== null) {
+            resolve(response.toObject())
+          }
+        },
+      )
+    })
   }
 
-  /**
-   * sendTxSyncMode
-   * @param data
-   */
-  async sendTxSyncMode(data: object): Promise<TransactionSyncMode> {
-    let response = await this.post('/txs', {
-      tx: data,
-      mode: 'sync',
+  async getReporters(validator: string): Promise<string[]> {
+    const request = new QueryReportersRequest()
+    request.setValidatorAddress(validator)
+    return new Promise((resolve, reject) => {
+      this.queryClient.reporters(
+        request,
+        {} as grpc.Metadata,
+        (err, response) => {
+          if (err !== null) {
+            reject(err)
+            return
+          }
+
+          if (response !== null && response) {
+            resolve(response.getReporterList())
+          }
+        },
+      )
     })
+  }
 
-    let code = 0
-    let errorLog
-
-    if (response.code) {
-      code = parseInt(response.code)
-      errorLog = response.raw_log
-    }
+  async getLatestBlock(): Promise<GetLatestBlockResponse.AsObject> {
+    const request = new GetLatestBlockRequest()
+    return new Promise((resolve, reject) => {
+      this.serviceClient.getLatestBlock(
+        request,
+        {} as grpc.Metadata,
+        (err, response) => {
+          if (err !== null) {
+            reject(err)
+            return
+          }
+
+          if (response !== null && response) {
+            resolve(response.toObject())
+          }
+        },
+      )
+    })
+  }
 
-    return {
-      txHash: Buffer.from(response.txhash, 'hex'),
-      code,
-      errorLog,
-    }
+  async getAccount(address: string): Promise<BaseAccount.AsObject> {
```

Comment:
I think this one should receive `Account` instance instead of string.

Reply:
In `QueryAccountRequest`, `setAddress` accept a parameter value of type string. If address is a type `Account`, conversion have to be occured.

Reply:
I think using string address should be OK now.